### PR TITLE
Add process builder

### PR DESCRIPTION
**Pull request by tklaback**

### CODE REVIEW

**Review by ericmercer — DISMISSED** [review]
Where are you catching the exceptions and turning them into `Error` types? Is that coming in a later PR?

File: src/bpmncwpverify/builder/process_builder.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        try:
+            self._bpmn.processes[self._process.id] = self._process
+            self._construct_flow_network()
+            from bpmncwpverify.visitors.process_connectivity_visitor import (
+                ProcessConnectivityVisitor,
+            )
+
+            visitor = ProcessConnectivityVisitor()
+            self._process.accept(visitor)
+        except Exception as e:
```
Comment (tklaback) [inline src/bpmncwpverify/builder/process_builder.py:144]: here, I am catching the error and returning it as a failure

File: src/bpmncwpverify/builder/cwp_builder.py [hunk 2]
```diff
@@ -70,35 +70,40 @@ def _add_and_check_expressions(
...
+                if id not in self._cwp.start_states
+            }
+
+            # This step ensures connectivity of the graph and sets leaf edges
+            visitor = CwpConnectivityVisitor()
+            self._cwp.accept(visitor)
+
+            return Success(self._cwp)
+        except Exception as e:
+            return Failure(e.args[0])
```
Comment (tklaback) [inline src/bpmncwpverify/builder/cwp_builder.py:106]: updated this in the cwp_builder as well. Maybe should have put this in another PR

**Review by ericmercer — APPROVED** [review]
BTW, I saw a naming convention for the builder pattern the moves away from the `add` language to a `with` language. So its `builder.with_message(...)`. Will you make a not to covert our builders once we have core functionality in place?

**Review by jminyoon — COMMENTED** [review]
Looks good to me
